tracing/tracing.sh

```diff
@@ -31,3 +31,145 @@ function sync_main_kw_file_with_tracing()
   printf '%s' "$main_kw_file_with_tracing" > "${bin}/${kw_path}"
   chmod +x "${bin}/${kw_path}"
 }
+
+# This function injects code that enables tracing in every kw library file (files
+# inside the `src` directory) and installs them. Files pertaining to Bash and Zsh
+# completions (`src/bash_autocomplete.sh` and `src/_kw`), and the `src/VERSION`
+# aren't injected.
+#
+# @input_dir: Path to directory where base library files are stored
+# @output_dir: Path to directory where altered files with tracing will be installed
+function sync_kw_lib_files_with_tracing()
+{
+  local input_dir="$1"
+  local output_dir="$2"
+  local file_with_tracing
+  local parent_directory
+  local all_filepaths
+  local tmp_dir_path
+
+  tmp_dir_path=$(mktemp --directory)
```

Comment:
Is it possible that mktemp fails?

Reply:
Thanks for the catch. However I couldn't think of a way that this specific `mktemp` command can fail. Differently from your other comment, where I setted a different base from the default (`mktemp --directory --tmp-dir=/dev/shm`), I could think of and reproduce fail cases.